processor/resourcedetectionprocessor/internal/kubeadm/kubeadm.go

```diff
@@ -0,0 +1,62 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package kubeadm // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal/kubeadm"
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/processor"
+	conventions "go.opentelemetry.io/collector/semconv/v1.6.1"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/metadataproviders/kubeadm"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal/kubeadm/internal/metadata"
+)
+
+const (
+	TypeStr = "kubeadm"
+)
+
+var _ internal.Detector = (*detector)(nil)
+
+type detector struct {
+	provider kubeadm.Provider
+	logger   *zap.Logger
+	ra       *metadata.ResourceAttributesConfig
+	rb       *metadata.ResourceBuilder
+}
+
+func NewDetector(set processor.Settings, dcfg internal.DetectorConfig) (internal.Detector, error) {
+	cfg := dcfg.(Config)
+	if err := cfg.UpdateDefaults(); err != nil {
+		return nil, err
+	}
+
+	k8snodeProvider, err := kubeadm.NewProvider(cfg.configMapName, cfg.configMapNamespace, cfg.APIConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed creating kubeadm provider: %w", err)
+	}
+
+	return &detector{
+		provider: k8snodeProvider,
+		logger:   set.Logger,
+		ra:       &cfg.ResourceAttributes,
+		rb:       metadata.NewResourceBuilder(cfg.ResourceAttributes),
+	}, nil
+}
+
+func (d *detector) Detect(ctx context.Context) (resource pcommon.Resource, schemaURL string, err error) {
+	if d.ra.K8sClusterName.Enabled {
+		clusterName, err := d.provider.ClusterName(ctx)
+		if err != nil {
+			return pcommon.NewResource(), "", fmt.Errorf("failed getting k8s cluster name: %w", err)
```

Comment:
Nit, feel free to drop: I'm wondering if it would be better to future-proof this by not returning the error right away, and returning the error at the end. 

If we end up adding more resource attributes the existing logic would not record the other, unrelated resource attributes, even if they could be detected and emitted successfully.